build-system/common/ci.js

```diff
@@ -15,121 +15,231 @@
+ */
+const isTravis = isTravisBuild();
+const isGithubActions = isGithubActionsBuild();
+const isCircleci = isCircleciBuild();
+
 /**
  * Returns true if this is a PR build.
  * @return {boolean}
  */
 function isPullRequestBuild() {
-  return serviceFunctionMap['isPullRequestBuild']();
+  return isTravis
+    ? env('TRAVIS_EVENT_TYPE') === 'pull_request'
+    : isGithubActions
+    ? env('GITHUB_EVENT_NAME') === 'pull_request'
+    : isCircleci
+    ? !!env('CIRCLE_PULL_REQUEST')
+    : false;
 }
 
 /**
  * Returns true if this is a push build.
  * @return {boolean}
  */
 function isPushBuild() {
-  return serviceFunctionMap['isPushBuild']();
+  return isTravis
+    ? env('TRAVIS_EVENT_TYPE') === 'push'
+    : isGithubActions
+    ? env('GITHUB_EVENT_NAME') === 'push'
+    : isCircleci
+    ? !env('CIRCLE_PULL_REQUEST')
+    : false;
 }
 
 /**
  * Returns the name of the PR branch.
  * @return {string}
  */
 function ciPullRequestBranch() {
-  if (!isPullRequestBuild()) {
-    log(red('ERROR:'), 'Not a PR build, PR branch is undefined.');
-    return '';
-  }
-  return serviceFunctionMap['ciPullRequestBranch']();
+  return isTravis
+    ? env('TRAVIS_PULL_REQUEST_BRANCH')
+    : isGithubActions
+    ? env('GITHUB_HEAD_REF') // TODO(rsimha): Truncate if necessary.
+    : isCircleci
+    ? env('CIRCLE_BRANCH')
+    : '';
 }
 
 /**
- * Returns the commit SHA being tested by the PR build.
+ * Returns the commit SHA being tested by a PR build.
  * @return {string}
  */
 function ciPullRequestSha() {
-  if (!isPullRequestBuild()) {
-    log(red('ERROR:'), 'Not a PR build, PR SHA is undefined.');
-    return '';
-  }
-  return serviceFunctionMap['ciPullRequestSha']();
+  return isTravis
+    ? env('TRAVIS_PULL_REQUEST_SHA')
+    : isGithubActions
+    ? env('GITHUB_SHA')
+    : isCircleci
+    ? env('CIRCLE_SHA1')
+    : '';
+}
+
+/**
+ * Returns the branch for push builds.
+ * @return {string}
+ */
+function ciPushBranch() {
+  return isTravis
+    ? env('TRAVIS_BRANCH')
+    : isGithubActions
+    ? env('GITHUB_REF') // TODO(rsimha): Truncate if necessary.
+    : isCircleci
+    ? env('CIRCLE_BRANCH')
+    : '';
+}
+
+/**
+ * Returns the commit SHA being tested by a push build.
+ * @return {string}
+ */
+function ciCommitSha() {
+  return isTravis
+    ? env('TRAVIS_COMMIT')
+    : isGithubActions
+    ? env('GITHUB_SHA')
+    : isCircleci
+    ? env('CIRCLE_SHA1')
+    : '';
+}
+
+/**
+ * Returns the number of the current build.
+ * @return {string}
+ */
+function ciBuildNumber() {
+  return isTravis
+    ? env('TRAVIS_BUILD_NUMBER')
+    : isGithubActions
+    ? env('GITHUB_RUN_ID')
+    : isCircleci
+    ? env('CIRCLE_BUILD_NUM')
+    : '';
+}
+
+/**
+ * Returns the URL of the current build.
+ * @return {string}
+ */
+function ciBuildUrl() {
+  return isTravis
+    ? env('TRAVIS_BUILD_WEB_URL')
+    : isGithubActions
+    ? `${env('GITHUB_SERVER_URL')}/${env('GITHUB_REPOSITORY')}/actions/runs/${env('GITHUB_RUN_ID')}` // prettier-ignore
```

Comment:
One option to keep the line width down, I'm undecided on if it's an improvement or not
```suggestion
    ? [
        env('GITHUB_SERVER_URL'),
        env('GITHUB_REPOSITORY'),
        'actions',
        'runs',
        env('GITHUB_RUN_ID'),
      ].join('/)
```